src/Features/Core/Portable/AddImport/IAddImportFeatureService.cs

```diff
@@ -16,5 +17,18 @@ internal interface IAddImportFeatureService : ILanguageService
             Document document, TextSpan span, string diagnosticId, bool placeSystemNamespaceFirst,
             ISymbolSearchService symbolSearchService, bool searchReferenceAssemblies,
             ImmutableArray<PackageSource> packageSources, CancellationToken cancellationToken);
+
+        Task<ImmutableArray<(Diagnostic Diagnostic, ImmutableArray<AddImportFixData> Fixes)>> GetFixesForDiagnosticsAsync(
+            Document document, TextSpan span, ImmutableArray<Diagnostic> diagnostics,
+            bool searchReferenceAssemblies, bool searchNuGetPackages, CancellationToken cancellationToken);
+
+        Task<ImmutableArray<(Diagnostic Diagnostic, ImmutableArray<AddImportFixData> Fixes)>> GetFixesForDiagnosticsAsync(
+            Document document, TextSpan span, ImmutableArray<Diagnostic> diagnostics,
+            ISymbolSearchService symbolSearchService, bool searchReferenceAssemblies,
+            ImmutableArray<PackageSource> packageSources, CancellationToken cancellationToken);
```

Comment:
who calls the first method versus the second?

Reply:
Removed first method and made each caller responsible for passing in the services.